File: C4 - Ciclos/PY/while.py
# Contador y suma inicializados en cero
suma = 0
contador = 0

# Ciclo 'while' que continúa mientras se ingresen números positivos
while True:
    numero = float(input("Ingresa un número positivo (o un número negativo para salir): "))
    
    if numero < 0:
        break  # Sale del ciclo si el número es negativo
    
    suma += numero  # Suma el número a la suma total
    contador += 1  # Incrementa el contador de números positivos ingresados

# Imprime el resultado final
print(f"Has ingresado {contador} números positivos.")
print(f"La suma de esos números es: {suma}")

File: C4 - Ciclos/C++/while.cpp
#include <iostream>
using namespace std;

int main() {
    char opcion; // Variable para almacenar la opción ingresada por el usuario

    // Ciclo 'while' que se ejecuta mientras el usuario no ingrese 'n' o 'N'
    while (true) {
        cout << "¿Quieres continuar? (s/n): ";
        cin >> opcion;

        // Verifica si el usuario ingresó 'n' o 'N' para salir del ciclo
        if (opcion == 'n' || opcion == 'N') {
            cout << "Has elegido salir del programa." << endl;
            break; // Termina el ciclo
        }

        // Si el usuario ingresa cualquier otra cosa, continúa el ciclo
        cout << "Has elegido continuar. ¡Sigamos!" << endl;
    }

    return 0;
}